### ISSUE TITLE
Bug: Does not support snapshot blocks

### ISSUE THREAD

**Issue by tconbeer** [tconbeer/sqlfmt#137]
**Describe the bug**
We support common jinja blocks, but we forgot about `{% snapshot my_snapshot %}`, so the `{% endsnapshot %}` tag can get put on the wrong line.

**To Reproduce**
```
{% snapshot snp_my_snapshot %}

{{
    config(
          target_database='analytics',
          target_schema=target.schema + '_snapshots',
          unique_key='id',
          strategy='timestamp',
          updated_at='updated_at',
        )
}}

select * from {{ ref('stg_my_model') }}
{% endsnapshot %}
```

**Expected behavior**
```
{% snapshot snp_my_snapshot %}

{{
    config(
          target_database='analytics',
          target_schema=target.schema + '_snapshots',
          unique_key='id',
          strategy='timestamp',
          updated_at='updated_at',
        )
}}

select * 
from {{ ref('stg_my_model') }}
{% endsnapshot %}
```

**Actual behavior**
```
{% snapshot snp_my_snapshot %}

{{
    config(
          target_database='analytics',
          target_schema=target.schema + '_snapshots',
          unique_key='id',
          strategy='timestamp',
          updated_at='updated_at',
        )
}}

select * 
from {{ ref('stg_my_model') }}{% endsnapshot %}
```

**Additional context**
What is the output of `sqlfmt --version`? 0.5.1

**Comment by HologramInterview** [comment]
We've seen this happening in our codebase, thanks for filing it.

**Comment by rdeese** [comment]
We've seen this happening in our codebase, thanks for filing it.

**Comment by tconbeer** [comment]
@rdeese The fix for this is now merged to main and will be included in the next patch release (hopefully end of this week)

**Comment by tconbeer** [comment]
@rdeese This has been released as v0.6.0 -- please upgrade for the patch
